### ISSUE TITLE
When _id is an ObjectID

### ISSUE THREAD

**Issue by darkship** [yeputons/meteor-smart-publish#6]
Hi,

I inserted  in my mongoDB a row ( so it was without Meteor.Collection) and it happened that mongoDB creates an <code>ObjectID</code>([meteor can do it too](http://docs.meteor.com/#collection_object_id)) instead of a <code>string</code> for <code>_id</code>.

And it looks like that smart-publish has some difficulties to handle it. So I removed that row and everything was fine.

```
Exception from sub u8TXuWx35cBTwEauk Error: Could not find element with id ObjectID("53fda8c3edcac19a2a8ee975") to change
    at Object.Future.wait (/var/www/upload/projectiz/bundle/programs/server/node_modules/fibers/future.js:326:15)
    at _.extend.runTask (packages/meteor/fiber_helpers.js:84)
    at _.extend.addHandleAndSendInitialAdds (packages/mongo-livedata/observe_multiplex.js:47)
    at MongoConnection._observeChanges (packages/mongo-livedata/mongo_driver.js:1108)
    at Cursor.observeChanges (packages/mongo-livedata/mongo_driver.js:783)
    at publishCursor (packages/smart-publish/smart-publish.js:171)
    at null.<anonymous> (packages/smart-publish/smart-publish.js:193)
    at null._handler (packages/collection-hooks/collection-hooks.js:199)
    at maybeAuditArgumentChecks (packages/livedata/livedata_server.js:1487)
    at _.extend._runHandler (packages/livedata/livedata_server.js:914)
    - - - - -
    at Object.Future.wait (/var/www/upload/projectiz/bundle/programs/server/node_modules/fibers/future.js:326:15)
    at _.extend.runTask (packages/meteor/fiber_helpers.js:84)
    at _.extend.addHandleAndSendInitialAdds (packages/mongo-livedata/observe_multiplex.js:47)
    at MongoConnection._observeChanges (packages/mongo-livedata/mongo_driver.js:1108)
    at Cursor.observeChanges (packages/mongo-livedata/mongo_driver.js:783)
    at publishCursor (packages/smart-publish/smart-publish.js:171)
    at packages/smart-publish/smart-publish.js:72
    at Function._.each._.forEach (packages/underscore/underscore.js:113)
    at updateChildren (packages/smart-publish/smart-publish.js:44)
    at smartAdded (packages/smart-publish/smart-publish.js:94)
    - - - - -
    at _.extend.changed (packages/livedata/livedata_server.js:180)
    at _.extend.changed (packages/livedata/livedata_server.js:392)
    at _.extend.changed (packages/livedata/livedata_server.js:1086)
    at updateFromData (packages/smart-publish/smart-publish.js:31)
    at smartRemoved (packages/smart-publish/smart-publish.js:135)
    at packages/smart-publish/smart-publish.js:77
    at Array.forEach (native)
    at packages/smart-publish/smart-publish.js:76
    at Function._.each._.forEach (packages/underscore/underscore.js:113)
    at updateChildren (packages/smart-publish/smart-publish.js:44)
    at smartAdded (packages/smart-publish/smart-publish.js:102)
```

**Comment by yeputons** [comment]
@darkship should be fixed in v0.1.8 , feel free to re-open if it is not.

**Comment by darkship** [comment]
Nice, I will try it later.

Thanks
